Plugin/Model/SavePlugin.php

```diff
@@ -79,4 +80,19 @@ public function afterSave(PageRepository $subject, Page $result)
 
         return $result;
     }
+
+    private function buildUrlFromPage(Page $page): string
+    {
+        if (!array_key_exists(0, $page->getStoreId())) {
+            return '';
+        }
+
+        try {
+            $storeFromPage = $this->storeManager->getStore($page->getStoreId()[0]);
```

Comment:
Just a suggestion:

```suggestion
            $storeIds = $page->getStoreId();
            if (!count($storeIds)) {
                return '';
            }
            
            try {
                $firstStoreId = $storeIds[0];
                $storeFromPage = $this->storeManager->getStore($firstStoreId);
                
                return $storeFromPage->getBaseUrl() . $page->getIdentifier();
            } catch (NoSuchEntityException $e) {
                return '';
            }
```

Reply:
Interesting suggestion! Why do you not explicitly check for index `0` to be in the output of `$page->getStoreId()`?

Reply:
Readability: it's not about specific key `0`, but rather about if the array is filled.